File: src/main/java/com/procurement/contracting/model/dto/ContractStatus.java
package com.procurement.contracting.model.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.HashMap;
import java.util.Map;

public enum ContractStatus {
    PENDING("pending"),
    ACTIVE("active"),
    CANCELLED("cancelled"),
    COMPLETE("complete"),
    TERMINATED("terminated");

    private static final Map<String, ContractStatus> CONSTANTS = new HashMap<>();

    static {
        for (final ContractStatus c : values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    private final String value;

    ContractStatus(final String value) {
        this.value = value;
    }

    @JsonCreator
    public static ContractStatus fromValue(final String value) {
        final ContractStatus constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new IllegalArgumentException(value);
        }
        return constant;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }
}

File: src/main/kotlin/com/procurement/contracting/domain/model/contract/status/ContractStatus.kt
package com.procurement.contracting.domain.model.contract.status

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import com.procurement.contracting.domain.model.EnumElementProvider

enum class ContractStatus(@JsonValue override val key: String) : EnumElementProvider.Key {
    PENDING("pending"),
    ACTIVE("active"),
    CANCELLED("cancelled"),
    COMPLETE("complete"),
    TERMINATED("terminated"),
    UNSUCCESSFUL("unsuccessful");

    override fun toString(): String = key

    companion object : EnumElementProvider<ContractStatus>(info = info()) {
        @JvmStatic
        @JsonCreator
        fun creator(name: String) = ContractStatus.orThrow(name)
    }
}
